### PR TITLE
Add explanation on how to work with private repos (DVC+MLEM)

### PR DESCRIPTION
**Pull request by aguschin**
https://mlem-ai-dvc-private-rep-yxphuh.herokuapp.com/doc/user-guide/dvc/

close https://github.com/iterative/mlem/issues/528

### CODE REVIEW

File: content/docs/user-guide/dvc.md [hunk 1]
```diff
@@ -147,3 +147,21 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
 
 Learn more about [DVC](https://dvc.org/doc) and how it can be useful for
 training your ML models.
+
+## Working with private repositories
+
+If you commit your models to a private repository and use DVC to store binaries,
```
Comment (jorgeorpinel) [inline content/docs/user-guide/dvc.md:153]:
```suggestion
If you commit model metafiles to a private repo and use DVC to store binaries,
```

File: content/docs/user-guide/dvc.md [hunk 2]
```diff
@@ -147,3 +147,21 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
...
+If you commit model metafiles to a private repo and use DVC to store binaries,
+you'll need to authorize both via SSH (for DVC) and via HTTPS (for MLEM).
+
+SSH authorization is usually achieved by running `git push` against a SSH
+remote, or it can be done with
+[`gh auth login`](https://cli.github.com/manual/gh_auth_login) (if you use
+Github).
+
+HTTPS authorization is done by setting `GITHUB_USERNAME` and `GITHUB_TOKEN` env
+vars. You can generate a token [here](https://github.com/settings/tokens).
```
Comment (omesser) [inline content/docs/user-guide/dvc.md:162]: env vars -> environment variabls
Also it's not clear here if you need or not to generate a token ("can" ?)

**Review by omesser — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/dvc.md [hunk 3]
```diff
@@ -147,3 +147,21 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
 
 Learn more about [DVC](https://dvc.org/doc) and how it can be useful for
 training your ML models.
+
+## Working with private repositories
+
+If you commit model metafiles to a private repo and use DVC to store binaries,
+you'll need to authorize both via SSH (for DVC) and via HTTPS (for MLEM).
```
Comment (omesser) [inline content/docs/user-guide/dvc.md:154]: authorize -> authenticate
Would be more accurate everywhere here imo

**Review by omesser — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/dvc.md (hunk 2 above)
Comment (aguschin) [reply content/docs/user-guide/dvc.md:162]: Yep, it's a limitation. I'll add few sentences.

Comment (aguschin) [reply content/docs/user-guide/dvc.md:162]: I don't really know why DVC doesn't work with HTTP authentication. Maybe @skshetry can briefly explain the reason.

File: content/docs/user-guide/dvc.md (hunk 3 above)
Comment (aguschin) [reply content/docs/user-guide/dvc.md:154]: Fixed.

File: content/docs/user-guide/dvc.md [hunk 4]
```diff
@@ -147,3 +147,26 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
...
 training your ML models.
+
+## Working with private repositories
+
+If you commit model metafiles to a private repo and use DVC to store binaries,
+you'll need to authenticate both via SSH and via HTTPS. SSH authentication is
+required for DVC, since DVC shallow clones the repo underneath via SSH. MLEM
+instead uses `fsspec`'s
+[`GitHubFileSystem`](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.github.GithubFileSystem)
+to access the repo, which uses HTTP and authenticate with it underneath.
```
Comment (omesser) [inline content/docs/user-guide/dvc.md:158]:
```suggestion
to access the repo, which uses HTTPS for authentication.
```

File: content/docs/user-guide/dvc.md [hunk 5]
```diff
@@ -147,3 +147,26 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
...
+
+SSH authentication is usually achieved by running `git push` against a SSH
+remote, or it can be done with
```
Comment (omesser) [inline content/docs/user-guide/dvc.md:161]:
```suggestion
remote, or can be done using
```

File: content/docs/user-guide/dvc.md [hunk 6]
```diff
@@ -147,3 +147,26 @@ metafile (`models/rf.mlem`) and `dvc.lock` only.
...
+[`gh auth login`](https://cli.github.com/manual/gh_auth_login) (if you use
+Github).
+
+HTTPS authentication is done by setting `GITHUB_USERNAME` and `GITHUB_TOKEN`
+environment variables. You need to generate a token
+[here](https://github.com/settings/tokens).
+
+It's is important to first authenticate with SSH and then with HTTPS. Otherwise,
```
Comment (omesser) [inline content/docs/user-guide/dvc.md:169]:
```suggestion
It's important to first authenticate with SSH, and only then with HTTPS. Otherwise,
```

**Review by omesser — CHANGES_REQUESTED** [review]
minor copy-editing 🙏

**Review by omesser — APPROVED** [review]

**Pull request merged**